src/party.rs

```diff
@@ -0,0 +1,90 @@
+//! Definition of the BPCon participant structure
+
+use std::sync::mpsc::{Receiver, Sender};
+use crate::{Value, ValueSelector};
+use crate::error::BallotError;
+use crate::message::{MessageRouting, ProtocolMessage};
+
+/// BPCon configuration. Includes ballot time bounds, and other stuff
+pub struct BallotConfig {
+    // TODO: define config fields
+}
+
+/// Party of the BPCon protocol that execute ballot.
+///
+/// The communication between party and external
+/// system is done via `in_receiver` and `out_sender` channels. External system should take
+/// care about authentication while receiving incoming messages and then push them to the
+/// corresponding `Sender` to the `in_receiver`. Same, it should tke care about listening of new
+/// messages in the corresponding `Receiver` to the `out_sender` and submitting them to the
+/// corresponding party based on information in `MessageRouting`.
+///
+/// After finishing of the ballot protocol, party will place the selected value to the
+/// `value_sender` or `BallotError` if ballot failed.
+pub struct Party<V: Value, VS: ValueSelector<V>> {
+    /// This party's identifier.
+    pub id: u64,
+    /// Other ballot parties' ids in a couple with weight.
+    pub party_ids: Vec<(u64, u128)>,
```

Comment:
I personally dislike tuples as they are non-descriptive, at least when dealing with non-named types like this (u64, u128) -- could we use a struct for the vec contents? Could even use a generic Tuple<A, B> type.

Reply:
I've certainly used tuples in these kinds of situations, but I generally prefer structs with good doc comments.  Looking at this code it's not obvious what weight represents.

Reply:
I guess maybe we'll move the weight into the `BallotConfig`, thus for now left only ids.